### PR TITLE
Fix mypy errors - part 1

### PR DESCRIPTION
**Pull request by e-larsson**
Part of #51

### CODE REVIEW

File: scanomatic/image_analysis/signal.py [hunk 1]
```diff
@@ -637,7 +639,7 @@ def get_closest_signal_pair(
 
     S = np.array(signals)
     if S.size == 0:
-        return None, None
+        raise SignalError("No signal pairs found")
```
Comment (e-larsson) [inline scanomatic/image_analysis/signal.py:642]: The None tuple wasn't handled anywhere

File: scanomatic/image_analysis/signal.py [hunk 2]
```diff
@@ -727,7 +729,7 @@ def get_offset_quality(
     return q.sum()
 
 
-def _get_wave_length_and_errors(s, expected_spikes: int) -> float:
```
Comment (e-larsson) [inline scanomatic/image_analysis/signal.py:730]: Unused parameter

Comment (skymandr) [reply scanomatic/image_analysis/signal.py:730]: I think these share an interface, hence why it has that argument, but @local-minimum can hopefully elucidate.

Comment (local-minimum) [reply scanomatic/image_analysis/signal.py:730]: I did a lot of yagni back then and probably used it while developing the algorithm to figure something out, but seems reasonable to remove now

File: scanomatic/image_analysis/signal.py [hunk 3]
```diff
@@ -661,7 +663,7 @@ def get_signal_spikes(
     ).astype(int)
 
     s1, s2 = get_closest_signal_pair(d_up, d_down, s1_value=-1, s2_value=1)
-    return (s1 + s2) / 2.0  # (s1 + s2) / 2.0
```
Comment (local-minimum) [inline scanomatic/image_analysis/signal.py:664]: what a great comment 😆

File: scanomatic/image_analysis/signal.py (hunk 1 above)
Comment (local-minimum) [reply scanomatic/image_analysis/signal.py:642]: sounds good consider getting rid of the `S` variable as something like `signals_arr` or something too.

**Review by local-minimum — APPROVED** [review]

**Pull request merged**